### PR TITLE
[AAP-7483] Add Execution Env Details

### PR DESCRIPTION
**Pull request by prat98**
This PR addresses [AAP-7483](https://issues.redhat.com/browse/AAP-7843) which is about adding the execution env details tab. This PR adds... 

- The details page tab component
- The parent tab page component which contains the details tab and templates tab
- Component tests for the details tab
- Missed component tests for the EE lists page
- Routing to get to the details tab ( `1/infrastructure/execution-environments/:id/details` implemented in this PR) and templates tab ( `1/infrastructure/execution-environments/4/templates` currently set as `<PageNotImplemented>` component)

### CODE REVIEW

**Review by vishsanghishetty — APPROVED** [review]
🚀

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+export function ExecutionEnvironmentDetailInner(props: { execution_env: ExecutionEnvironment }) {
+  const { t } = useTranslation();
+  const getPageUrl = useGetPageUrl();
+  const pageNavigate = usePageNavigate();
+
+  const execution_env = props.execution_env;
+
+  return (
+    <PageDetails data-cy="execution-environment-page-detail">
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Name')}>
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:38]: can this data-cy value be 'execution-environment-name' instead? I see that 'execution-environment-page-detail is repeated several times in this file- Cypress will get confused and not know which one to click on if we use it as a locator. They all need to be unique.

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        {execution_env.name}
+      </PageDetail>
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Image')}>
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:41]: can this data-cy value be changed to 'execution-environment-image'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        {execution_env.image}
+      </PageDetail>
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Description')}>
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:44]: can this data-cy value be changed to 'execution-environment-description'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        {execution_env.description}
+      </PageDetail>
+      <PageDetail
+        data-cy="execution-environment-page-detail"
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:48]: can this data-cy be changed to 'execution-environment-managed'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        label={t('Managed')}
+      >{`${execution_env.managed}`}</PageDetail>
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Organization')}>
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:51]: can this data-cy be changed to 'execution-environment-org'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        {execution_env.summary_fields?.organization ? (
+          <TextCell
+            text={execution_env.summary_fields?.organization?.name}
+            to={getPageUrl(AwxRoute.OrganizationPage, {
+              params: { id: execution_env.summary_fields?.organization?.id },
+            })}
+          />
+        ) : undefined}
+      </PageDetail>
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Pull')}>
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:61]: can this data-cy be changed to 'execution-environment-pull'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        {execution_env.pull}
+      </PageDetail>
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Registry Credential')}>
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:64]: can this data-cy be changed to 'execution-environment-reg-cred'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        {execution_env?.summary_fields?.credential?.name ? (
+          <Label variant="outline" color="blue">
+            {execution_env?.summary_fields?.credential?.name}
+          </Label>
+        ) : undefined}
+      </PageDetail>
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Created')}>
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:71]: can this data-cy be changed to 'execution-environment-created'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+          author={execution_env?.summary_fields?.created_by?.username}
+          onClick={() =>
+            pageNavigate(AwxRoute.UserDetails, {
+              params: { id: execution_env?.summary_fields?.created_by?.id },
+            })
+          }
+        />
+      </PageDetail>
+      <LastModifiedPageDetail
+        data-cy="execution-environment-page-detail"
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:84]: can this data-cy be changed to 'execution-environment-modified'?

File: frontend/awx/administration/execution-environments/ExecutionEnvironments.cy.tsx [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+          method: 'GET',
+          url: '/api/v2/execution_environments/*',
+        },
+        {
+          statusCode: 500,
+        }
+      );
+      cy.mount(<ExecutionEnvironments />);
+      cy.contains('Error loading execution environments');
+    });
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironments.cy.tsx:222]: this test needs a .should assertion. 
.Contains by itself is not an assertion

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.cy.tsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  it('Component renders and displays Execution Environment', () => {
+    cy.fixture('execution_environments.json')
+      .then((eeResponse: AwxItemsResponse<ExecutionEnvironment>) => {
+        const execution_env: ExecutionEnvironment = eeResponse?.results[0];
+        return execution_env;
+      })
+      .then((eeObject) => {
+        cy.mount(<ExecutionEnvironmentDetails execution_env={eeObject} />);
+      });
+  });
```
Comment (akus062381) [inline frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.cy.tsx:16]: please add a .should assertion to the above test. 
.Contains by itself is not an assertion

**Review by akus062381 — CHANGES_REQUESTED** [review]

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx (hunk 1 above)
Comment (prat98) [reply frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:38]: done

File: frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx (hunk 2 above)
Comment (prat98) [reply frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx:41]: done

**Review by akus062381 — APPROVED** [review]

**Pull request merged**